corehq/middleware.py

```diff
@@ -336,3 +336,12 @@ def get_view_func(view_fn, view_kwargs):
         return view_fn.view_class
 
     return view_fn
+
+
+class SecureCookiesMiddleware(MiddlewareMixin):
+
+    def process_response(self, request, response):
+        if hasattr(response, 'cookies') and response.cookies:
+            for cookie in response.cookies:
+                response.cookies[cookie]['secure'] = settings.SECURE_COOKIES
```

Comment:
Should this be
```suggestion
                response.cookies[cookie]['secure'] = response.cookies[cookie]['secure'] or settings.SECURE_COOKIES
```
That is, force it to secure if `SECURE_COOKIES`, otherwise do whatever the caller specified.  As-is, this will prohibit any cookie setting from being secure if `not SECURE_COOKIES`.

Though as you said, we are exclusively using HTTPS, so maybe we shouldn't even bother checking `settings.SECURE_COOKIES`, and just always use `True`

Reply:
I think we want to continue to use the `settings.SECURE_COOKIES` to allow other envs to specify in case they aren't serving over HTTPS? But you make a good point, and I was unsure of the best behavior but wrote this test to reflect this case:
https://github.com/dimagi/commcare-hq/blob/0f550f957614cfef756539bfb8d1042e09148529/corehq/tests/test_middleware.py#L225-L228

I could easily be convinced to always lean towards securing a cookie, so if it enters this middleware already secured, leave it that way regardless of `settings.SECURE_COOKIES`. I think that makes more sense so I'll change that.

Reply:
See f947d34e5f2730c6b0084c655b35efe26ac10349

Reply:
Cool yeah, those tests are for sure helpful - that test was what made me think about this aspect of the behavior.